buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestTestPlugin.java

```diff
@@ -0,0 +1,122 @@
+                SOURCE_SET_NAME,
+                RestIntegTestTask.class,
+                task -> { task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask")); }
+            );
+        yamlRestTestTask.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+        yamlRestTestTask.setDescription("Runs the YAML based REST tests against an external cluster");
+
+        // setup task dependency
+        if (BuildParams.isInternal()) {
+            project.getDependencies().add(SOURCE_SET_NAME + "Compile", project.project(":test:framework"));
+        } else {
+            project.getDependencies()
+                .add(SOURCE_SET_NAME + "Compile", "org.elasticsearch.test:framework:" + VersionProperties.getElasticsearch());
+        }
+
+        // setup the runner
+        RestTestRunnerTask runner = (RestTestRunnerTask) project.getTasks().getByName(yamlRestTestTask.getName() + "Runner");
+        runner.setTestClassesDirs(yamlTestSourceSet.getOutput().getClassesDirs());
+        runner.setClasspath(yamlTestSourceSet.getRuntimeClasspath());
+
+        // if this a module or plugin, it may have an associated zip file with it's contents, add that to the test cluster
+        boolean isModule = project.getPath().startsWith(":modules:");
+        Zip bundle = (Zip) project.getTasks().findByName("bundlePlugin");
+        if (bundle != null) {
+            yamlRestTestTask.dependsOn(bundle);
+            if (isModule) {
+                runner.getClusters().forEach(c -> c.module(bundle.getArchiveFile()));
+            } else {
+                runner.getClusters().forEach(c -> c.plugin(project.getObjects().fileProperty().value(bundle.getArchiveFile())));
+            }
+        }
+
+        // es-plugins may declare dependencies on additional modules, add those to the test cluster too.
+        project.afterEvaluate(p -> {
+            PluginPropertiesExtension pluginPropertiesExtension = project.getExtensions().findByType(PluginPropertiesExtension.class);
+            if (pluginPropertiesExtension != null) { // not all projects are defined as plugins
+                pluginPropertiesExtension.getExtendedPlugins().forEach(pluginName -> {
+                    Project extensionProject = project.getProject().findProject(":modules:" + pluginName);
+                    if (extensionProject != null) { // extension plugin may be defined, but not required to be a module
+                        Zip extensionBundle = (Zip) extensionProject.getTasks().getByName("bundlePlugin");
+                        yamlRestTestTask.dependsOn(extensionBundle);
+                        runner.getClusters().forEach(c -> c.module(extensionBundle.getArchiveFile()));
+                    }
+                });
+            }
+        });
+
+        // setup IDE
+        GradleUtils.setupIdeForTestSourceSet(project, yamlTestSourceSet);
+
+        // make the new test run after unit tests
+        yamlRestTestTask.mustRunAfter(project.getTasks().named("test"));
```

Comment:
Can we move this up where we configure other aspects of this task?

Reply:
done [`6289b00` (#56841)](https://github.com/elastic/elasticsearch/pull/56841/commits/6289b00b7fb71856968e2a6fa543202f4aa8cf7d)